thunder/core/module.py

```diff
@@ -187,7 +187,10 @@ def __getattr__(self, name: str) -> Any:
 
     def state_dict(self, *args: Any, **kwargs: Any) -> Any:
         # this is broken for transformed modules!!!
-        return self._model.state_dict(*args, **kwargs)
+        state_dict = self._model.state_dict(*args, **kwargs)
```

Comment:
This should probably stat with `self.state_dict()` and then reverse transform it.
I would love to have code like this, but in a method `original_state_dict`.